### ISSUE TITLE
DocumentSymbol ranges do not appear to abide by protocol requirements

### ISSUE THREAD

**Issue by leungbk** [microsoft/pyright#1592]
**Describe the bug**
Given a location for the insertion point, the symbol displayed in the headerline does not always align with expectations.

**To Reproduce**
With the Pyright extension for VSCode, open the following Python file:

``` python
def hi():
    return 42


def yo():
    return 42
```

When the insertion point is here:

``` python
d[*]ef yo():
```

the headerline displays the `hi` symbol.

When the insertion point is here:

``` python
de[*]f yo():
```

the headerline displays the `yo` symbol.

The relevant DocumentSymbols and their (zero-indexed) ranges are

``` elisp
(
 (:name "hi" :kind 12 :range (:START (:line 0 :character 0) :END (:line 4 :character 1)) :selectionRange (:start (:line 0 :character 4) :end (:line 0 :character 6)) :children #1=[])

 (:name "yo" :kind 12 :range (:START (:line 4 :character 0) :END (:line 5 :character 14)) :selectionRange (:start (:line 4 :character 4) :end (:line 4 :character 6)) :children #1#))
```
**Expected behavior**
The protocol says of DocumentSymbol.range

> The range enclosing this symbol not including leading/trailing whitespace but everything else like comments. This information is typically used to determine if the clients cursor is inside the symbol to reveal in the symbol in the UI.

Based on this, I would have expected the DocumentSymbol.range for the `hi` function to end after the first `return 42` but strictly before `def yo():`. For both of the starred locations for the insertion point above, I would have expected the header line to display `yo`.

**VS Code extension or command-line**
VS Code extension 1.1.115

**Comment by erictraut** [comment]
It looks like you're running an old version of the extension. This was fixed in 1.1.118.

**Comment by leungbk** [comment]
I see, thanks!
